src/transform/src/analysis/equivalences.rs

```diff
@@ -366,79 +407,36 @@ impl EquivalenceClasses {
         while !stable {
             stable = self.minimize_once(&columns);
         }
-
-        // TODO: remove these measures once we are more confident about idempotence.
-        let prev = self.clone();
-        self.minimize_once(&columns);
-        mz_ore::soft_assert_eq_or_log!(self, &prev, "Equivalences::minimize() not idempotent");
```

Comment:
I'm feeling a bit uneasy about removing this check just in the same PR where the stability check in `minimize_once` is also being changed.

Reply:
Ok, this is fine after checking the new stability check in more detail, see https://github.com/MaterializeInc/materialize/pull/30082#discussion_r1808915374